packages/generators/addon-generator.ts

```diff
@@ -24,54 +26,57 @@ const copyUtils = require("@webpack-cli/utils/copy-utils");
  *
  * @returns {Generator} A class extending Generator
  */
-function addonGenerator(
-	prompts,
-	templateDir,
-	copyFiles,
-	copyTemplateFiles,
-	templateFn
+export default function addonGenerator(
+	prompts: IScaffoldBaseObject[],
+	templateDir: string,
+	copyFiles: string[],
+	copyTemplateFiles: string[],
+	templateFn: Function,
 ) {
-	//eslint-disable-next-line
-	return class extends Generator {
-		prompting() {
-			return this.prompt(prompts).then(props => {
+	return class AddOnGenerator extends Generator {
+		public props: IScaffoldBaseObject;
+		private copy: (value: string, index: number, array: string[]) => void;
+		private copyTpl: (value: string, index: number, array: string[]) => void;
+
+		public prompting() {
+			return this.prompt(prompts).then((props: IScaffoldBaseObject) => {
 				this.props = props;
 			});
 		}
 
-		default() {
-			const currentDirName = path.basename(this.destinationPath());
+		public default() {
+			const currentDirName: string = path.basename(this.destinationPath());
 			if (currentDirName !== this.props.name) {
 				this.log(`
 				Your project must be inside a folder named ${this.props.name}
 				I will create this folder for you.
 				`);
-				mkdirp(this.props.name);
-				const pathToProjectDir = this.destinationPath(this.props.name);
+				mkdirp(this.props.name, (err: object) => {
```

Comment:
missing ret type

Reply:
We aren't storing whatever `mkdirp` returns then why and where to add types?